systemtest/fleet_test.go

```diff
@@ -0,0 +1,159 @@
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/transport"
+
+	"github.com/elastic/apm-server/systemtest"
+	"github.com/elastic/apm-server/systemtest/apmservertest"
+	"github.com/elastic/apm-server/systemtest/fleettest"
+)
+
+func TestFleetIntegration(t *testing.T) {
+	systemtest.CleanupElasticsearch(t)
+
+	fleet := fleettest.NewClient(systemtest.KibanaURL.String())
+	require.NoError(t, fleet.Setup())
+	cleanupFleet(t, fleet)
+	defer cleanupFleet(t, fleet)
+
+	agentPolicy, enrollmentAPIKey, err := fleet.CreateAgentPolicy("apm_systemtest", "default", "Agent policy for APM Server system tests")
+	require.NoError(t, err)
+
+	// Find the "apm" package to install.
+	var apmPackage *fleettest.Package
+	packages, err := fleet.ListPackages()
+	require.NoError(t, err)
+	for _, pkg := range packages {
+		if pkg.Name == "apm" {
+			apmPackage = &pkg
+			break
+		}
+	}
+	require.NotNil(t, apmPackage)
+
+	// Add the "apm" integration to the agent policy.
+	packagePolicy := fleettest.PackagePolicy{
+		Name:          "apm",
+		Namespace:     "default",
+		Enabled:       true,
+		AgentPolicyID: agentPolicy.ID,
+	}
+	packagePolicy.Package.Name = apmPackage.Name
+	packagePolicy.Package.Version = apmPackage.Version
+	packagePolicy.Package.Title = apmPackage.Title
+	packagePolicy.Inputs = []fleettest.PackagePolicyInput{{
+		Type:    "apm",
+		Enabled: true,
+		Streams: []interface{}{},
+		Vars: map[string]interface{}{
+			"enable_rum": map[string]interface{}{
+				"type":  "bool",
+				"value": false,
+			},
+			"host": map[string]interface{}{
+				"type":  "string",
+				"value": ":8200",
+			},
+		},
+	}}
+	err = fleet.CreatePackagePolicy(packagePolicy)
+	require.NoError(t, err)
+
+	agent, err := systemtest.NewUnstartedElasticAgentContainer()
+	require.NoError(t, err)
+	agent.FleetEnrollmentToken = enrollmentAPIKey.APIKey
+
+	// Build apm-server, and bind-mount it into the elastic-agent container's "install"
+	// directory. This bypasses downloading the artifact.
+	arch := runtime.GOARCH
+	if arch == "amd64" {
+		arch = "x86_64"
+	}
+	apmServerArtifactName := fmt.Sprintf("apm-server-%s-linux-%s", agent.StackVersion, arch)
+
+	// Bind-mount the apm-server binary and apm-server.yml into the container's
+	// "install" directory. This causes elastic-agent to skip installing the
+	// artifact.
+	apmServerBinary, err := apmservertest.BuildServerBinary("linux")
+	require.NoError(t, err)
+	agent.BindMountInstall[apmServerBinary] = path.Join(apmServerArtifactName, "apm-server")
+	apmServerConfigFile, err := filepath.Abs("../apm-server.yml")
+	require.NoError(t, err)
+	agent.BindMountInstall[apmServerConfigFile] = path.Join(apmServerArtifactName, "apm-server.yml")
+
+	// Start elastic-agent with port 8200 exposed, and wait for the server to service
+	// healthcheck requests to port 8200.
+	agent.ExposedPorts = []string{"8200"}
+	waitFor := wait.ForHTTP("/")
+	waitFor.Port = "8200/tcp"
+	agent.WaitingFor = waitFor
+	require.NoError(t, agent.Start())
```

Comment:
I'd prefer if `agent.Start()` call happens in a separated line

Reply:
done